### PR TITLE
Mqtt getting started guide

### PR DESCRIPTION
**Pull request by 2m**

### CODE REVIEW

File: mqtt/src/test/scala/akka/stream/contrib/mqtt/MqttSourceSpec.scala [hunk 1]
```diff
@@ -54,6 +53,46 @@ class MqttSourceSpec extends WordSpec with Matchers with ScalaFutures {
...
+      import f._
+
+      //#create-settings
+      val settings = MqttSourceSettings(
+        MqttConnectionSettings(
+          "tcp://localhost:1883",
+          "test-client",
+          new MemoryPersistence
+        ),
+        Map("topic1" -> 0, "topic2" -> 0)
```
Comment (johanandren) [inline mqtt/src/test/scala/akka/stream/contrib/mqtt/MqttSourceSpec.scala]: what are these numbers?

Comment (johanandren) [inline mqtt/src/test/scala/akka/stream/contrib/mqtt/MqttSourceSpec.scala]: "These zeroes needs to be here, trust me"

File: mqtt/src/main/scala/Mqtt.scala [hunk 2]
```diff
@@ -11,6 +11,19 @@ import org.eclipse.paho.client.mqttv3.{ MqttMessage => PahoMqttMessage, _ }
...
 
+/**
+ * Quality of Service constants as defined in
+ * https://www.eclipse.org/paho/files/mqttdoc/Cclient/qos.html
+ */
+object QoS {
+  def atMostOnce = 0
+  def atLeastOnce = 1
+  def exactlyOnce = 2
+}
```
Comment (johanandren) [inline mqtt/src/main/scala/Mqtt.scala]: Better. But I would have liked a typed encoding even better. As in sealed trait + three objects (with a byte value method each for the internal representation).

**Comment by 2m** [comment]
Added typed `MqttQoS`. Its a sealed abstract *class* because only then one can access a method defined in `MqttQoS` objecy from Java like: `MqttQoS.atMostOnce();`.

File: mqtt/src/test/scala/akka/stream/contrib/mqtt/MqttSourceSpec.scala [hunk 3]
```diff
@@ -54,7 +53,47 @@ class MqttSourceSpec extends WordSpec with Matchers with ScalaFutures {
...
+      import f._
+
+      //#create-settings
+      val settings = MqttSourceSettings(
+        MqttConnectionSettings(
+          "tcp://localhost:1883",
+          "test-client",
+          new MemoryPersistence
+        ),
+        Map("topic1" -> MqttQoS.AtMostOnce, "topic2" -> MqttQoS.AtMostOnce)
```
Comment (johanandren) [inline mqtt/src/test/scala/akka/stream/contrib/mqtt/MqttSourceSpec.scala]: 100 times much nicer, thanks!

**Pull request merged**